src/queue.rs

```diff
@@ -517,13 +532,13 @@ impl<M: GuestAddressSpace> Queue<M> {
     }
 
     /// Puts an available descriptor head into the used ring for use by the guest.
-    pub fn add_used(&mut self, desc_index: u16, len: u32) {
+    pub fn add_used(&mut self, desc_index: u16, len: u32) -> Option<u16> {
         if desc_index >= self.actual_size() {
             error!(
                 "attempted to add out of bounds descriptor to used ring: {}",
                 desc_index
             );
-            return;
+            return None;
```

Comment:
Returning an `Option` from this method doesn't feel quite right, since if we're ever in a position to return `None`, then we've encountered a serious error. I'm not sure what's the best way to handle various `Queue` error conditions (this look like another discussion to be had post this PR), but for now I think this method should either return a `Result<u16>`, or directly  an `u16` that's equal to `self.next_used.0 - 1` (the standard says that, with `VIRTIO_F_EVENT_IDX` negotiated, a notification must be sent when the `used_idx` we've just written to was equal to `used_event`).